common:
  base_link_name: torso_lift_link
  end_effector_link_name: l_wrist_roll_link
  force_torque_frame: l_gripper_tool_frame
  solver: # parameters used for the IkSolverVel algorithm
    epsilon: 0.0001
    alpha: 0.5
    maxiter: 200
    nso_weights: 1
  feedback_rate: 50
  force_torque_topic: /ft_sensor/ft_compensated

manipulation_controller:
  action_server_name: dexterous_manipulation
  spring_constant: 0.17
  gains:
    k_1: 0.7
    k_2: 1.0
    k_3: 3.2
  saturations:
    x: 0.1
    y: 0.1
    theta: 0.05
  rotational_gains: [10, 0.0, 10] #[5, 0, 5] # These should be IN THE SURFACE FRAME
  estimate_length: false
  hardcoded_length: 0.11 # from the gripper tool frame to the part tip
  initial_angle_offset: -0.0
  wait_for_tf_time: 1
  surface_frame_vertical_offset: -0.01
  surface_frame_horizontal_offset: 0.02
  estimator:
    Q:
      data: [100, 0, 0, 0, 10, 0, 0, 0, 4]
    R:
      data: [0.01, 0, 0, 0, 0.01, 0, 0, 0, 0.002]

    initial_x_offset: -0.15
    initial_theta_offset: 0

approach_controller:
  action_server_name: approach_controller
  rotational_gains: [5, 5, 5]
  contact_detection_time: 0.1

move_controller:
  action_server_name: move_controller
  max_allowed_error: 0.5
  velocity_gain: 1.0
  error_threshold: 0.01
  ik_service_name: /pr2_left_arm_kinematics/get_ik
  ik_info_service_name: /pr2_left_arm_kinematics/get_ik_solver_info
